deepchem/models/torch_models/layers.py

```diff
@@ -23,6 +23,173 @@
+      same value is used for every layer, 'relu' by default
+    mode: str
+      Either 'classification' or 'regression'
+    residual: bool
+      if True, the model will be composed of pre-activation residual blocks instead
+      of a simple stack of linear layers.
+    ode_block: nn.Module
+      whether to use neural ode
+
+    """
+
+    self.n_tasks = n_tasks
+    self.droputs = dropouts
+    self.ode_block = ode_block
+    self.n_classes - n_classes
+    self.residual = residual
+    self.mode = mode
+    self.enable_ffn = enable_ffn
+
+    n_layers = len(layer_filters)
+
+    if len(layer_filters) == 1:
+      layer_filters = layer_filters * 2
+    if not isinstance(dropouts, SequenceCollection):
+      dropouts = [dropouts] * n_layers
+    if isinstance(activation_fns,
+                  str) or not isinstance(activation_fns, SequenceCollection):
+      activation_fns = [activation_fns] * n_layers
+    if not isinstance(weight_init_stddevs, SequenceCollection):
+      weight_init_stddevs = [weight_init_stddevs] * n_layers
+    if not isinstance(bias_init_consts, SequenceCollection):
+      bias_init_consts = [bias_init_consts] * n_layers
+
+    self.activation_fns = [get_activation(f) for f in activation_fns]
+
+    self.net = nn.ModuleList()
+
+    in_shape = n_features
+
+    for out_shape, weight_stddev, bias_const in zip(layer_filters,
+                                                    weight_init_stddev,
+                                                    bias_init_consts):
+
+      layer = nn.Linear(in_features=in_shape, out_features=out_shape, bias=True)
+
+      nn.init.normal_(layer.weight, 0, weight_stddev)
+
+      if layer.bias is not None:
+        layer.bias = nn.Parameter(torch.full(layer.bias.shape, bias_const))
+
+      self.layer.append(layer)
+
+    self.classifier_ffn = nn.LazyLinear(self.n_tasks * self.n_classes)
+    self.output_layer = nn.LazyLinear(self.n_tasks)
+
+  def forward(self, inputs):
+    """
+    Parameters
+    ----------
+    inputs: list of torch.Tensor
+      Input Tensors
+    Returns
+    -------
+    torch.Tensor
+      Output Tensor
+    """
+
+    if isinstance(inputs, torch.Tensor):
+      x, dropout_switch = inputs, None
+    else:
+      x, dropout_switch = inputs
+
+    prev_layer = x
+
+    for layer, activation_fn, dropout in zip(self.layers, self.activation_fns,
+                                             self.dropouts):
+      x = layer(x)
+
+      if dropout > 0. and dropout_switch:
+        x = F.dropout(x, dropout)
+
+        if self.residual and x.shape[1] == prev_layer.shape[1]:
+          x = x + prev_layer
+
+        if activation_fn is not None:
+          x = self.activation_fn(x)
+
+        prev_layer = x
+
+    if self.ode_block is not None:
+      x, _ = self.ode_block(x)
+
+    outputs = x
+
+    if self.enable_ffn:
+      batch_size = x.shape[0]
+
+      if self.mode == "classification":
```

Comment:
A linear layer shouldn't know about classification/regression

Reply:
encapsulating the vector field layer within neuralode model
[reference](https://github.com/deepchem/deepchem/blob/master/deepchem/models/fcnet.py)